x/auth/mapper.go

```diff
@@ -116,6 +130,26 @@ func (am AccountMapper) setSequence(ctx sdk.Context, addr sdk.Address, newSequen
 	return nil
 }
 
+// Returns and increments the global account number counter
+func (am AccountMapper) GetNextAccountNumber(ctx sdk.Context) int64 {
+	var accNumber int64
+	store := ctx.KVStore(am.key)
+	bz := store.Get(globalAccountNumberKey)
+	if bz == nil {
```

Comment:
This works, but you can also set the first account number in `InitGenesis` - a bit cleaner.

Reply:
InitGenesis calls the accountmapper NewAccount to create the genesis accounts anyways, so the initial globalAccountNumber will always be 0.

Reply:
Do we need the conditional then? `bz` should never be nil